src/Microsoft.AspNet.Mvc.Core/Filters/FormatFilterAttribute.cs

```diff
@@ -0,0 +1,125 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.AspNet.Mvc.Description;
+using Microsoft.Framework.DependencyInjection;
+using Microsoft.Framework.OptionsModel;
+using Microsoft.Net.Http.Headers;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNet.Mvc
+{
+    /// <summary>
+    /// A filter which will use the format value in the route data or query string to set the content type on an 
+    /// <see cref="ObjectResult" /> returned from an action.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method, AllowMultiple = false, Inherited = true)]
+    public class FormatFilterAttribute : Attribute, IFormatFilter, IResourceFilter, IResultFilter
+    {
+        /// <summary>
+        /// As a <see cref="IResourceFilter"/>, this filter looks at the request and rejects it before going ahead if
+        /// 1. The format in the request doesnt match any format in the map.
+        /// 2. If there is a conflicting producesFilter.
+        /// </summary>
+        /// <param name="context">The <see cref="ResourceExecutingContext"/>.</param>
+        public void OnResourceExecuting([NotNull] ResourceExecutingContext context)
+        {
+            var format = GetFormat(context);
+
+            if (!string.IsNullOrEmpty(format))
+            {
+                var formatContentType = GetContentType(format, context);
+                if (formatContentType == null)
+                {
+                    // no contentType exists for the format, return 404
+                    context.Result = new HttpNotFoundResult();
+                }
+                else
+                {
+                    var responseTypeFilters = context.Filters.OfType<IApiResponseMetadataProvider>();                    
+                    var contentTypes = new List<MediaTypeHeaderValue>();
+
+                    foreach (var filter in responseTypeFilters)
+                    {
+                        filter.SetContentTypes(contentTypes);
+                    }
+
+                    if (contentTypes.Count() != 0)
```

Comment:
Try to avoid ever using `Count()` it's generally a code smell because it means that you're working with an `IEnumerable` and you're going to iterate it twice, which we generally don't want to do. If you ever find a genuine need for `.Count()` there is almost always a better choice.

In this case use `.Count` instead (the property).